src/universal/components/BaseButton.js

```diff
@@ -0,0 +1,129 @@
+import PropTypes from 'prop-types'
+import React, {Component} from 'react'
+import styled from 'react-emotion'
+import ui from 'universal/styles/ui'
+import PlainButton from 'universal/components/PlainButton/PlainButton'
+
+const ButtonRoot = styled(PlainButton)(({size, depth, disabled, pressedDown}) => {
+  const hasDepth = depth || depth === 0
+  const boxShadow = hasDepth ? ui.shadow[depth] : 'none'
+  const hoverDepth = hasDepth ? ui.shadow[depth + 2] : ui.shadow[1]
+  const pressedDepth = hasDepth ? ui.shadow[depth + 1] : ui.shadow[0]
+  const stateDepth = pressedDown ? pressedDepth : hoverDepth
+  return {
+    // size is easy to override, it adds: fontSize, lineHeight, padding
+    ...ui.buttonSizeStyles[size],
+    display: 'block',
+    border: '.0625rem solid transparent',
+    boxShadow,
+    textAlign: 'center',
+    transform: pressedDown && 'translate(0, .125rem)',
+    transition: `
+        box-shadow ${ui.buttonTransition},
+        transform ${ui.buttonTransition}
+      `,
+    userSelect: 'none',
+    whiteSpace: 'nowrap',
+    ':hover,:focus,:active': {
+      boxShadow: disabled ? 'none' : stateDepth,
+      outline: pressedDown && 0
+    }
+  }
+})
+
+class BaseButton extends Component {
+  static propTypes = {
+    'aria-label': PropTypes.string,
+    size: PropTypes.oneOf(ui.buttonSizeOptions),
+    children: PropTypes.any,
+    // handling className allows usage of emotion’s styled handler
+    className: PropTypes.string,
+    // depth: up to 2 + 2 (for :hover, :focus) = up to ui.shadow[4]
+    depth: PropTypes.oneOf([0, 1, 2]),
+    disabled: PropTypes.bool,
+    innerRef: PropTypes.func,
+    onClick: PropTypes.func,
+    onMouseEnter: PropTypes.func,
+    onMouseLeave: PropTypes.func,
+    style: PropTypes.object,
+    waiting: PropTypes.bool
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      pressedDown: null
+    }
+  }
+
+  onMouseDown = (e) => {
+    if (e.button === 0) {
+      this.setState({pressedDown: true})
+    }
+  }
+
+  onMouseUp = (e) => {
+    if (this.state.pressedDown) {
+      this.setState({pressedDown: false})
+    }
+    // We don’t want 'focus' styles to linger after the click (TA)
+    // wait till next tick because other components might need to use the button as the relativeTarget when they get blurred
+    // pull the target out of the event so react can recycle the event
+    const {currentTarget} = e
```

Comment:
`currentTarget` is consistently the `button` (event is attached to the button) whereas `target` is the thing clicked that triggered the event, like an inner element such as a `span` that handles some icon. When the `button` just has a text node as children then both `currentTarget` and `target` are the same. I got in the habit over the years of using `currentTarget` because calling the `target` wasn’t as consistent based on the behaviors desired.